packages/ember-runtime/lib/computed/array_computed.js

```diff
@@ -163,9 +163,11 @@ ArrayComputedProperty.prototype.didChange = function (obj, keyName) {
   @param {String} [dependentKeys*]
   @param {Object} options
   @return {Ember.ComputedProperty}
+  @deprecated
   @private
 */
 function arrayComputed(options) {
+  Ember.deprecate('Ember.arrayComputed is deprecated. Replace it with plain array methods');
```

Comment:
Can you move this into the constructor [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/computed/array_computed.js#L12) and [here](https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/computed/reduce_computed.js#L478)? 

Might need a flag to make sure that the deprecation is for ReduceComputed or ArrayComputed...